### PR TITLE
PHP 8 Support

### PR DESCRIPTION
**Pull request by j03k64**
# PHP 8 Support
This PR adds support for getting this library working on PHP 8. I'm going to leave this open for a bit while I determine if there are other places that need to be addressed as well.

## Issue 1
`get_resource_type` fails with `$this->socket` passed in on PHP8 since it's expecting a `resource` instead of a `Socket` object.

### Fix 1
Update `Net/Gearman/Connection.php` to return early if `$this->socket` is an instance of `Socket` (which should be only on PHP8+), otherwise falls back to default logic.

## Issue 2
Unable to run functional tests.

### Fix 2
- Updated `ConnectionTest` and `TaskTest` to use more modern code including short array syntax and the `assertArray` method introduced in phpunit 7.5. This required a bump of the minimum phpunit to 7.5 from 7.3.
- Updated `ConnectionTest` to check for `PHP_VERSION >= 8.0` and if so check for the `Socket` object instead of `socket` string.
- Added a `tests/README.md` for how to run the unit tests.

### CODE REVIEW

**Comment by j03k64** [comment]
@brianlmoon thoughts on how to update the unit test? Do a PHP version check like below:
```php
// Net_Gearman_ConnectionTest.php:19
if (version_compare($var, '8.0.0') >= 0) {
  // PHP 8+ returns a Socket class instead of a resource now
  $this->assertInstanceOf('Socket', $connections->socket);
}
else {
  $this->assertEquals('socket', strtolower(get_resource_type($connection->socket)));
}

or would you prefer something else?

**Comment by brianlmoon** [comment]
@j03k64 Makes sense to me to do it that way.

**Comment by j03k64** [comment]
@brianlmoon I don't have an easy way to test this on a PHP 7.X install. I could figure it out, but if you have something readily available that would be appreciated.

File: Net/Gearman/Connection.php [hunk 1]
```diff
@@ -476,11 +476,21 @@ public function close()
...
+            return true;
+        }
+
         // HHVM returns stream. PHP 5.x returns socket
         $type = strtolower(get_resource_type($this->socket));
-        return (is_null($this->socket) !== true &&
-                is_resource($this->socket) === true &&
-                ($type == 'socket' || $type == "stream"));
+
+        return is_resource($this->socket) === true
```
Comment (brianlmoon) [inline Net/Gearman/Connection.php:492]: Seems like we should maybe move the `is_resource() check` above the `get_resource_type()` check. What do you think? Default the function to return false at the end.

```php
if(is_resource($this->socket) === true) {
    $type = strtolower(get_resource_type($this->socket));
    return ($type == 'socket' || $type == 'stream');
}

return false;
```

File: Net/Gearman/Connection.php [hunk 2]
```diff
@@ -476,11 +476,21 @@ public function close()
...
+            && ($type == 'socket' || $type == 'stream');
```
Comment (brianlmoon) [inline Net/Gearman/Connection.php:493]: HHVM no longer supports PHP, so afaik, we can remove the 'stream' check now.

Comment (j03k64) [reply Net/Gearman/Connection.php:493]: Cool, I'll simplify that check.

**Comment by j03k64** [comment]
@brianlmoon any other comments? (I have this branch in production running on 7.4 fwiw)

**Review by brianlmoon — APPROVED** [review]

**Pull request merged**